### ISSUE TITLE
Create TypeScript typings

### ISSUE THREAD

**Issue by Wedvich** [structured-log/structured-log#39]
For TypeScript applications that use the library, a `.d.ts` typings file should be available through the `@types` repository for easier consumption.

**Comment by sstorie** [comment]
Is this something open for a pull request? 

I'm using this library to create custom sinks in an Angular (2+) project, and would like to leverage the types in my own code. I could take a stab at building the index.d.ts file if the author is up for maintaining it over time.

**Comment by Wedvich** [comment]
@sstorie Yep, this is absolutely open for a PR! And having the initial set of types will make it easier to maintain as the library evolves as well.

**Comment by sstorie** [comment]
Ok, I've created typings for another project, but it's been a few months so let me dust off the brain and try to get something to you in the next week or so.

**Comment by davideicardi** [comment]
I usually just set `"declaration": true` in `tsconfig.json`, add `"types": "./lib/index.d.ts"` in package.json and then let `tsc` do the rest. But seems that using rollup is not so easy. Or I don't know how to do it...

What is the reason to use rollup instead of just tsc? To bundle files? Maybe you can use tsc to compile and then rollup to bundle?

**Comment by andymac4182** [comment]
This would be great as even though its typescript code its hard to work with in another typescript application. https://github.com/jaredpalmer/formik might provide some hints as where to look at it appears they use rollup and produce typescript definitions.

**Comment by RoryDungan** [comment]
I've created a pull request for this #49

**Comment by biliktamas79** [comment]
+1
